src/course-libraries/ReviewTabContent.tsx

```diff
@@ -0,0 +1,377 @@
+            </Stack>
+          </Stack>
+          {actions}
+        </Stack>
+      </Card.Section>
+    </Card>
+  );
+};
+
+const ComponentReviewList = ({
+  outOfSyncComponents,
+  onSearchUpdate,
+}: {
+  outOfSyncComponents: PublishableEntityLink[];
+  onSearchUpdate: () => void;
+}) => {
+  const intl = useIntl();
+  const { showToast } = useContext(ToastContext);
+  const [blockData, setBlockData] = useState<LibraryChangesMessageData | undefined>(undefined);
+  // ignore changes confirmation modal toggle.
+  const [isConfirmModalOpen, openConfirmModal, closeConfirmModal] = useToggle(false);
+  const {
+    hits: downstreamInfo,
+    isLoading: isIndexDataLoading,
+    searchKeywords,
+    searchSortOrder,
+    hasError,
+  } = useSearchContext() as {
+    hits: ContentHit[];
+    isLoading: boolean;
+    searchKeywords: string;
+    searchSortOrder: SearchSortOption;
+    hasError: boolean;
+  };
+  const outOfSyncComponentsByKey = useMemo(
+    () => _.keyBy(outOfSyncComponents, 'downstreamUsageKey'),
+    [outOfSyncComponents],
+  );
+  const downstreamInfoByKey = useMemo(
+    () => _.keyBy(downstreamInfo, 'usageKey'),
+    [downstreamInfo],
+  );
+  const queryClient = useQueryClient();
+
+  useEffect(() => {
+    if (searchKeywords) {
+      onSearchUpdate();
+    }
+  }, [searchKeywords]);
+
+  // Toggle preview changes modal
+  const [isModalOpen, openModal, closeModal] = useToggle(false);
+  const acceptChangesMutation = useAcceptLibraryBlockChanges();
+  const ignoreChangesMutation = useIgnoreLibraryBlockChanges();
+
+  const setSeletecdBlockData = (info: ContentHit) => {
+    setBlockData({
+      displayName: info.displayName,
+      downstreamBlockId: info.usageKey,
+      upstreamBlockId: outOfSyncComponentsByKey[info.usageKey].upstreamUsageKey,
+      upstreamBlockVersionSynced: outOfSyncComponentsByKey[info.usageKey].versionSynced,
+      isVertical: info.blockType === 'vertical',
+    });
+  };
+  // Show preview changes on review
+  const onReview = (info: ContentHit) => {
+    setSeletecdBlockData(info);
+    openModal();
+  };
+
+  const onIgnoreClick = (info: ContentHit) => {
+    setSeletecdBlockData(info);
+    openConfirmModal();
+  };
+
+  const reloadLinks = (usageKey: string) => {
+    const courseKey = outOfSyncComponentsByKey[usageKey].downstreamContextKey;
+    queryClient.invalidateQueries(courseLibrariesQueryKeys.courseLibraries(courseKey));
+  };
+
+  const postChange = (accept: boolean) => {
+    // istanbul ignore if: this should never happen
+    if (!blockData) {
+      return;
+    }
+    reloadLinks(blockData.downstreamBlockId);
+    if (accept) {
+      showToast(intl.formatMessage(
+        messages.updateSingleBlockSuccess,
+        { name: blockData.displayName },
+      ));
+    } else {
+      showToast(intl.formatMessage(
+        messages.ignoreSingleBlockSuccess,
+        { name: blockData.displayName },
+      ));
+    }
+  };
+
+  const updateBlock = async (info: ContentHit) => {
```

Comment:
Since these functions like `updateBlock` are not using `useCallback`, they'll be re-created every time this component is re-rendered, e.g. every time the `searchKeywords` change. That may be totally fine, but it could also cause slow performance when typing search keywords on a page with many libraries. Keep an eye on it.

Reply:
Makes sense. Updated